src/test/java/io/reactivex/internal/operators/flowable/FlowableLimitTest.java

```diff
@@ -0,0 +1,208 @@
+/**
+ * Copyright (c) 2016-present, RxJava Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is
+ * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See
+ * the License for the specific language governing permissions and limitations under the License.
+ */
+
+package io.reactivex.internal.operators.flowable;
+
+import static org.junit.Assert.*;
+
+import java.util.*;
+
+import org.junit.Test;
+import org.reactivestreams.Subscriber;
+
+import io.reactivex.*;
+import io.reactivex.exceptions.*;
+import io.reactivex.functions.*;
+import io.reactivex.internal.subscriptions.BooleanSubscription;
+import io.reactivex.plugins.RxJavaPlugins;
+import io.reactivex.processors.PublishProcessor;
+import io.reactivex.subscribers.TestSubscriber;
+
+public class FlowableLimitTest implements LongConsumer, Action {
+
+    final List<Long> requests = Collections.synchronizedList(new ArrayList<Long>());
+
+    @Override
+    public void accept(long t) throws Exception {
+        requests.add(t);
+    }
+
+    @Override
+    public void run() throws Exception {
+        requests.add(-100L);
```

Comment:
super nit: `List<Object>`, `private static final Object CANCELLED = new Object()`, `requests.add(CANCELLED)`

```java
assertEquals(5, requests.get(0));
assertEquals(CANCELLED, requests.get(1));
```

Reply:
-100 is fine.